Dockerfile

```diff
@@ -1,46 +1,64 @@
-    go install github.com/golang/protobuf/protoc-gen-go && \
-    cd ../../ && rm -rf golang
-
-# Clone gogo/protobuf repo reset to the latest tag and install
-RUN mkdir gogo && cd gogo && git clone https://github.com/gogo/protobuf.git && \
-    cd protobuf && git reset --hard $(git rev-list --tags --max-count=1) && \
-    go install github.com/gogo/protobuf/protoc-gen-gogofast && \
-    go install github.com/gogo/protobuf/protoc-gen-gogofaster && \
-    go install github.com/gogo/protobuf/protoc-gen-gogoslick && \
-    cd ../../ && rm -rf gogo
-
-# Install the latest version of the protoc-gen-govalidator plugin
-RUN go get github.com/mwitkow/go-proto-validators/protoc-gen-govalidators && \
-    rm -rf mwitkow
-
-# Install the latest version of the protoc-gen-uwpartner plugin
-RUN go get github.com/utilitywarehouse/protoc-gen-uwpartner && \
-    rm -rf utilitywarehouse
-
-# Clone golangci repo reset to the latest tag and install
-RUN mkdir golangci && cd golangci && git clone https://github.com/golangci/golangci-lint.git && \
-    cd golangci-lint && git reset --hard $(git rev-list --tags --max-count=1) && \
-    go install github.com/golangci/golangci-lint/cmd/golangci-lint && \
-    cd ../../ && rm -rf golangci
+FROM golang:1.12.4
+
+ENV PROTOBUF_VERSION="3.7.0" \
+    GOLANG_PROTOBUF_VERSION="1.3.1" \
+    GOGO_PROTOBUF_VERSION="1.2.1" \
+    VALIDATE_PROTOBUF_VERSION="0.0.14" \
+    VALIDATORS_PROTOBUF_VERSION="1f388280e944c97cc59c75d8c84a704097d1f1d6" \
+    UWPARTNER_PROTOBUF_VERSION="de4552500027969912fd801dcc5269a153b3fffe" \
+    GOLANGCI_LINT_VERSION="1.16.0" \
+    DOCKER_VERSION="18.09.5"
+
+## Dependencies
+RUN apt-get update \
+    && apt-get install -y ca-certificates curl unzip tar
+
+## `protoc` binary
+RUN mkdir -p /tmp/protoc && cd /tmp/protoc \
+    && curl -sSL --retry 3 https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip -o protoc.zip \
+    && unzip protoc.zip \
+    && mv /tmp/protoc/bin/protoc* /usr/local/bin \
+    && mv /tmp/protoc/include/* /usr/local/include \
+    && rm -rf /tmp/protoc
+
+## `protoc-gen-go` binaries
+## `protoc-gen-gogoslick` binaries
+RUN GO111MODULE=on go get \
+    github.com/golang/protobuf/protoc-gen-go@v${GOLANG_PROTOBUF_VERSION} \
+    github.com/gogo/protobuf/protoc-gen-gogoslick@v${GOGO_PROTOBUF_VERSION} \
+    && mv /go/bin/protoc-gen-* /usr/local/bin/
+
+## `protoc-gen-validate` binary
+## `protoc-gen-govalidators` binary
+RUN GO111MODULE=on go get \
+    github.com/envoyproxy/protoc-gen-validate@v${VALIDATE_PROTOBUF_VERSION} \
+    github.com/mwitkow/go-proto-validators/protoc-gen-govalidators@${VALIDATORS_PROTOBUF_VERSION} \
+    && mv /go/bin/protoc-gen-* /usr/local/bin/
+
+## `protoc-gen-uwpartner` binary
+RUN GO111MODULE=on go get \
+    github.com/utilitywarehouse/protoc-gen-uwpartner@${UWPARTNER_PROTOBUF_VERSION} \
+    && mv /go/bin/protoc-gen-* /usr/local/bin/
```

Comment:
It would be nice to clean up modules cache after doing this as there is still `170M` of files in the `	/go/pkg/mod` directory